Mathlib/Data/BitVec.lean

```diff
@@ -76,6 +74,7 @@ theorem _root_.Fin.intCast_def' {n : Nat} [NeZero n] (x : Int) :
   dsimp [Int.cast, IntCast.intCast, Int.castDef]
   split <;> (simp [Fin.intCast]; omega)
 
+open Fin.CommRing in
 @[simp] theorem _root_.Fin.val_intCast {n : Nat} [NeZero n] (x : Int) :
```

Comment:
I don't understand why there is a *new* `open ...  in` here. Can you explain why it's necessary now but wasn't before?

Reply:
This was there before, because of the `open CommRing` on line 54. The change here is to make the `open` more specific.

Reply:
Indeed, this was to make clear which declarations are actually using the scope; in particular, whether the _type_ of the declaration needs the scope, or just the proof.

Reply:
Oh I see, yes, this is better.